lametro/templates/lametro/event.html

```diff
@@ -17,8 +17,8 @@ <h1>
                     {{event.name}}
                 {% endif%}
 
-                {% for media in event.media_urls.all %}
-                    <a class="btn btn-salmon" href="{{ media.url }}" target="_blank"><i class="fa fa-headphones" aria-hidden="true"></i> {{ media.label }}</a>
+                {% for media in event.media.all %}
+                    <a class="btn btn-salmon" href="{{ media.links.all.0.url }}" target="_blank"><i class= "fa fa-headphones" aria-hidden="true"></i> {% if media.note == 'Audio (SAP)' %}Ver en Español{% else %}Watch in English{% endif %}</a>
```

Comment:
I preferred having this functionality defined as a property of the media model. What was your rationale for moving it into the template?

Reply:
prefetching.

Reply:
Hm, I'm a little confused why returning a proxy model (LAMetroEventMedia) instead of the base model (EventMedia) interferes with the prefetching in this instance, since we aren't applying any additional filters to the queryset. What have I missed?  🙃

Reply:
it's not about the proxy model.

its that we have prefetch media and media links, If I did something like

`media.url` that would probably look something like this.

```python
class LAMetroEventMedia(...):
    def url(self):
        return self.links.get().url
```

calling that method would trigger a new query, and I wouldn't get the advantage of prefetching.

Reply:
Ah! I was referring to the label display logic, not the URL reference. Sorry for the confusion!!

Reply:
oh, it seemed like pure display logic. 

I suppose something like

`media.is_spanish` could be a bit nicer than `media.note == 'Audio(SAP)'`

Reply:
or event

`media.language == 'Spanish'`

Reply:
I liked the original property, which returned the display label, because you could make one change to update the label everywhere.

https://github.com/datamade/la-metro-councilmatic/blob/5f650c07c4a4ec665e6f3afc3284efa6cee50f90/lametro/models.py#L493-L502

What do you think?

Reply:
i think that model's shouldn't be responsible for display logic.

Reply:
Ok, how about a template tag then?

Reply:
the previous model attribute was only used in two templates. I think the clarity of where the display is defined is worth this level of duplication.

Reply:
Ok. I'm happy to agree to disagree on this minor point. No change required here.